### PR TITLE
AK: Mark Error::from_ functions as [[nodiscard]]

### PR DESCRIPTION
**Pull request by nico**
Prevents mistakes like the one fixed in #16672.

### CODE REVIEW

**Comment by trflynn89** [comment]
Good call, looks like this found the same error in Userland/Libraries/LibCompress/Deflate.cpp.

Marking the PR as "blocked" for now as it won't build anyways until #16672 is merged, just waiting for CI to finish.

**Comment by Hendiadyoin1** [comment]
Can't we mark the whole class as [[nodiscard]]?

**Comment by nico** [comment]
Sure, can do that in a follow-up. (Might need more cleanups.)

**Comment by trflynn89** [comment]
Other PR is merged, if you wanna rebase so we can see this pass

**Review by trflynn89 — APPROVED** [review]

**Pull request merged**
